plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererUtil.kt

```diff
@@ -161,32 +166,60 @@ object CodeWhispererUtil {
     }
 
     // This will be called only when there's a CW connection, but it has expired(either accessToken or refreshToken)
-    // 1. If connection is expired, try to refresh
-    // 2. If not able to refresh, requesting re-login by showing a notification
-    // 3. The notification will be shown
-    //   3.1 At most once per IDE restarts.
-    //   3.2 At most once after IDE restarts,
-    //   for example, when user performs security scan or fetch code completion for the first time
-    // Return true if need to re-auth, false otherwise
+    // 1. Attempt to refresh the connection
+    // 2. If refresh fails due to network issues (SdkClientException), it will:
```

Comment:
im still not understanding the need to do this while the user doesn't have network access